### PR TITLE
feat: event on error

### PR DESCRIPTION
**Pull request by alaibe**
Use event to sent signal as connection error

DO NOT MERGE until desktop is ready

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

File: services/wallet/thirdparty/cryptocompare/client.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func (c *Client) DoQuery(url string) (*http.Response, error) {
 }
 
 func (c *Client) FetchPrices(symbols []string, currencies []string) (map[string]map[string]float64, error) {
-	chunks := utils.ChunkSymbols(symbols)
+	chunks := utils.ChunkSymbols(symbols, 60)
```
Comment (Khushboo-dev-cpp) [inline services/wallet/thirdparty/cryptocompare/client.go:54]: why is this 60? just for me to understand

**Review by Khushboo-dev-cpp — APPROVED** [review]
LGTM :)

File: services/wallet/thirdparty/cryptocompare/client.go (hunk 1 above)
Comment (alaibe) [reply services/wallet/thirdparty/cryptocompare/client.go:54]: there is a limit on the api regarding the number of character, and I counted 5 character per symbol: 5*60 => 300

File: services/wallet/reader.go [hunk 2]
```diff
@@ -210,13 +197,17 @@ func (r *Reader) GetWalletToken(ctx context.Context, addresses []common.Address)
...
-		clients, err := r.rpcClient.EthClients(chainIDs)
-		if err != nil {
-			return err
-		}
 		balances, err = r.tokenManager.GetBalancesByChain(ctx, clients, addresses, tokenAddresses)
 		if err != nil {
+			for _, client := range clients {
+				client.SetIsConnected(false)
+			}
+			log.Info("tokenManager.GetBalancesByChain err", err)
```
Comment (dlipicar) [inline services/wallet/reader.go:210]: Do we want to keep this?

File: rpc/chain/client.go [hunk 3]
```diff
@@ -118,6 +122,28 @@ func isVMError(err error) bool {
...
+				c.WalletNotifier(c.ChainID, "down")
+			}
+			c.IsConnected = false
+		}
+
+	} else {
+		if !c.IsConnected {
+			c.IsConnected = true
+			if c.WalletNotifier != nil {
+				c.WalletNotifier(c.ChainID, "up")
```
Comment (dlipicar) [inline rpc/chain/client.go:141]: These "up" and "down" strings are defined in several different places. Could we have constants defined in a single spot?

File: services/wallet/reader.go (hunk 2 above)
Comment (dlipicar) [reply services/wallet/reader.go:210]: (And have it with log level `Info`)

File: services/wallet/thirdparty/cryptocompare/client.go (hunk 1 above)
Comment (dlipicar) [reply services/wallet/thirdparty/cryptocompare/client.go:54]: A constant defined at the top would be clearer

File: services/wallet/reader.go (hunk 2 above)
Comment (alaibe) [reply services/wallet/reader.go:210]: yeah, similar to other places, we can tweak later

File: rpc/chain/client.go (hunk 3 above)
Comment (alaibe) [reply rpc/chain/client.go:141]: I ll update with constants

**Pull request merged**